gossip/handler.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"context"
+	"time"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/network/p2p"
+	"github.com/ethereum/go-ethereum/log"
+	bloomfilter "github.com/holiman/bloomfilter/v2"
+)
+
+var _ p2p.Handler = (*Handler[Gossipable])(nil)
+
+func NewHandler[T Gossipable](set Set[T], codec codec.Manager, codecVersion uint16) *Handler[T] {
+	return &Handler[T]{
+		set:          set,
+		codec:        codec,
+		codecVersion: codecVersion,
+	}
+}
+
+type Handler[T Gossipable] struct {
+	set          Set[T]
+	codec        codec.Manager
+	codecVersion uint16
+}
+
+func (h Handler[T]) AppRequest(_ context.Context, nodeID ids.NodeID, _ time.Time, requestBytes []byte) ([]byte, error) {
```

Comment:
Can we add a comment upstream to make it clear that an error returned from this is not treated as fatal since it will be dropped by the p2p sdk?

Reply:
1. I didn't document this because i felt like an implementation shouldn't have to worry about how it's error is treated by the caller (unless something crazy happens like we handle it as a fatal).
2. I don't think this introduces a bug because we never return an error in any handling of any message. AFAICT we always log, which makes sense as I guess the question would be "when would we ever want to fatal when handling a message from a peer". I think we should pretty much always only be logging if something goes wrong.